src/optimistic-oracle/optimistic-oracle.spec.js

```diff
@@ -0,0 +1,389 @@
+const ethers = require('ethers');
+const { getAbi, getAddress } = require('@uma/contracts-node');
+const {
+  Finding, FindingSeverity, FindingType, createTransactionEvent,
+} = require('forta-agent');
+const { provideHandleTransaction, createPriceFeed } = require('./optimistic-oracle');
+
+// load functions from event manipulation library
+const { createLog, createReceipt } = require('../event-utils');
+
+// load agent configuration
+const {
+  umaEverestId,
+  optimisticOracle: optimisticOracleConfig,
+} = require('../../agent-config.json');
+
+const {
+  disputePriceErrorPercent,
+  cryptowatchApiKey: CRYPTOWATCH_API_KEY,
+  defipulseApiKey: DEFIPULSE_API_KEY,
+  tradermadeApiKey: TRADERMADE_API_KEY,
+  cmcApiKey: CMC_API_KEY,
+} = optimisticOracleConfig;
+
+// constant values
+const CHAIN_ID = 1; // mainnet
+const ZERO_ADDRESS = ethers.constants.AddressZero;
+
+// "BTC-BASIC-3M/USDC" identifier encoded as a Bytes32 value
+const BTC_IDENTIFIER_STRING = 'BTC-BASIS-3M/USDC';
+const BTC_IDENTIFIER_DATA = '0x4254432d42415349532d334d2f55534443000000000000000000000000000000';
+
+// "BAD" identifier encoded as a Bytes32 value
+const BAD_IDENTIFIER_DATA = '0x4241440000000000000000000000000000000000000000000000000000000000';
+
+const MOCK_PRICE = '99999999999999999999';
+
+const optimisticOracleAbi = getAbi('OptimisticOracle');
+
+// create interface
+const iface = new ethers.utils.Interface(optimisticOracleAbi);
+
+// mock the getPrice() function from optimistic-oracle.js
+async function mockGetPrice(identifier) {
+  if (identifier === BTC_IDENTIFIER_STRING) {
+    return MOCK_PRICE;
+  }
+
+  throw new Error(`Unknown identifier ${identifier}`);
+}
+
+async function mockGetPriceBadResponse(identifier) {
+  throw new Error(`Timeout or bad response to price feed request for identifier ${identifier}`);
+}
+
+describe('UMA optimistic oracle validation agent', () => {
+  let handleTransaction;
+  let optimisticOracleAddress = null;
```

Comment:
Looks like `optimistic-oracle.js` does checks on `optimisticOracleAddress` being `undefined` whereas we set it to `null` in the test code. Should one of these change so they're consistent?